Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -338,6 +340,17 @@ void ComplexTextController::collectComplexTextRuns()
     // would need to be updated accordingly too.
     nextFont = m_font.fontForCombiningCharacterSequence(baseOfString.first(currentIndex));
 
+    auto getHalfWidthFontIfNeeded = [&] (RefPtr<const Font>& protectedHalfWidthFont, TextSpacing::CharactersData& charactersData) {
+        if (m_font.textSpacingTrim().shouldTrimSpacing(charactersData))
+            protectedHalfWidthFont = nextFont->halfWidthFont();
+        return protectedHalfWidthFont ? protectedHalfWidthFont : nextFont;
```

Comment:
Doesn't it seem somewhat misleading for something called `getHalfWidthFontIfNeeded` to return something other than a pointer to a half width font? If we return `nextFont` does that represent some form of an error? Are we better off just returning a `nullptr` in this case and having the caller respond to that? Otherwise how can the caller distinguish between whether or not they got the half width font back?